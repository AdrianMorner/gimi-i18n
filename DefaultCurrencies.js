// @flow
export default {
  'SE': 'SEK',
  'NO': 'NOK',
  'DK': 'DKK',
  'GB': 'GBP',
  'US': 'USD',
  'FI': 'EUR',
  'SV': 'USD',
  'PA': 'USD',
  'EC': 'USD',
  'GU': 'USD',
  'PW': 'USD',
  'PR': 'USD',
  'ZW': 'USD',
  'AU': 'AUD',
  'CA': 'CAD',
  'NZ': 'NZD',
  'IN': 'INR',
  'ZA': 'ZAR',
  'NA': 'NAD',
  'BW': 'BWP',
  'SZ': 'SZL',
  'LS': 'LSL',
  'ZM': 'ZMF',
  'TZ': 'TZS',
  'KE': 'KES',
  'UG': 'UGX',
  'NG': 'NGN',
  'CM': 'XAF',
  'GH': 'GHS',
  'LR': 'LRD',
  'SL': 'SLL',
  'GM': 'GMD',
  'SC': 'SCR',
  'MU': 'MUR',
  'SH': 'SHP'
}
