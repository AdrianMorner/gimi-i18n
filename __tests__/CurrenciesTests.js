import {getSupportedCurrencyCodes, getDefaultCurrencyCode} from '../index'
import DefaultCurrencies from '../DefaultCurrencies'
import ExchangeRates from '../ExchangeRates'

jest.disableAutomock()

describe('Currencies', () => {
  it('it should return supported supportedCurrencyCodes', () => {
    expect(getSupportedCurrencyCodes()).not.toEqual(undefined)
    expect(getSupportedCurrencyCodes().length).toEqual(28)
    expect(getSupportedCurrencyCodes()).toMatchSnapshot()
  })

  it('it should return default currency', () => {
    expect(getDefaultCurrencyCode('FI')).toEqual('EUR')
    expect(getDefaultCurrencyCode('NO')).toEqual('NOK')
    expect(getDefaultCurrencyCode('GB')).toEqual('GBP')
    expect(getDefaultCurrencyCode('SE')).toEqual('SEK')
    expect(getDefaultCurrencyCode('BE')).toEqual('EUR')
    expect(getDefaultCurrencyCode()).toEqual('EUR')
  })

  Object.keys(DefaultCurrencies).forEach(key => {
    it(`Should be an exchange rate for ${key} (entry found in default currencies)`, () => {
      expect(ExchangeRates[DefaultCurrencies[key]]).toBeDefined()
    })
  })
})
