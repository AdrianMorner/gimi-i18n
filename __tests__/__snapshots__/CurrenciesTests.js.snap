exports[`Currencies it should return supported supportedCurrencyCodes 1`] = `
Array [
  "SEK",
  "NOK",
  "DKK",
  "GBP",
  "USD",
  "EUR",
  "AUD",
  "CAD",
  "NZD",
  "INR",
  "ZAR",
  "NAD",
  "BWP",
  "SZL",
  "LSL",
  "ZMF",
  "TZS",
  "KES",
  "UGX",
  "NGN",
  "XAF",
  "GHS",
  "LRD",
  "SLL",
  "GMD",
  "SCR",
  "MUR",
  "SHP",
]
`;
